native/core/src/execution/planner.rs

```diff
@@ -474,6 +481,123 @@ impl PhysicalPlanner {
+                                    }
+                                    DataType::Timestamp(TimeUnit::Microsecond, None) => {
+                                        let vals = values.clone();
+                                        SingleRowListArrayBuilder::new(Arc::new(TimestampMicrosecondArray::new(vals.long_values.into(), Some(vals.null_mask.into()))))
+                                            .build_list_scalar()
+                                    }
+                                    DataType::Timestamp(TimeUnit::Microsecond, Some(tz)) => {
+                                        let vals = values.clone();
+                                        SingleRowListArrayBuilder::new(Arc::new(TimestampMicrosecondArray::new(vals.long_values.into(), Some(vals.null_mask.into())).with_timezone(Arc::clone(tz))))
+                                            .build_list_scalar()
+                                    }
+                                    DataType::Date32 => {
+                                        let vals = values.clone();
+                                        SingleRowListArrayBuilder::new(Arc::new(Date32Array::new(vals.int_values.into(), Some(vals.null_mask.into()))))
+                                            .build_list_scalar()
+                                    }
+                                    DataType::Binary => {
+                                        // Using a builder here as it is quite complicated to create StringArray from a vector with nulls
+                                        // to calculate correct offsets
+                                        let vals = values.clone();
+                                        let len = vals.bytes_values.len();
+                                        let mut arr = BinaryBuilder::with_capacity(len, len);
+
+                                        for (i, v) in vals.bytes_values.into_iter().enumerate() {
+                                            if vals.null_mask[i] {
+                                                arr.append_value(v);
+                                            } else {
+                                                arr.append_null();
+                                            }
+                                        }
+
+                                        SingleRowListArrayBuilder::new(Arc::new(arr.finish()))
+                                            .build_list_scalar()
+                                    }
+                                    DataType::Utf8 => {
+                                        // Using a builder here as it is quite complicated to create StringArray from a vector with nulls
+                                        // to calculate correct offsets
+                                        let vals = values.clone();
+                                        let len = vals.string_values.len();
+                                        let mut arr = StringBuilder::with_capacity(len, len);
```

Comment:
Maybe add a TODO that there could be a better heuristic for the second arg than `len` or maybe even serialize over the number of bytes _a priori_.